### DISCUSSION TITLE
Select component dropdown does not work with gorhom bottom sheet

### DISCUSSION THREAD

**Discussion by saimarshadsaim31** [founded-labs/react-native-reusables#288] [Q&A] [answered]
I am using a select component in my project; overall, it works fine in normal places. But I also have a use case in which I want to use a select component inside of a bottom sheet. I am using the Gorhom bottom sheet in my project. Now the issue I am facing is that when I click the select option the dropdown appears underneath the bottom sheet I have tried to give styles like larger z index but the dropdown always renders below bottom sheet and when I close the bottom sheet it only stays there for a little bit and then disappears. Can some help me solve this???

**Comment by mrzachnugent** [comment, marked as answer]
Can you try [this pattern](https://rnr-docs.vercel.app/getting-started/common-patterns/#a-portal-component-inside-a-presentation-modal-screen)?

If that doesn't work, the issue might be where you place the `PortalHost` in your root layout. It should not be the child of the bottom-sheet provider.

**Reply by saimarshadsaim31** [reply]
Hey @mrzachnugent, thanks for the reply. It was indeed a dumb mistake on my end. This was the problem:
```
 <BottomSheetModalProvider>
          <StatusBar style={isDarkColorScheme ? 'light' : 'dark'} />
          <Stack>
            <Stack.Screen name="(tabs)" options={{ headerShown: false }} />
            <Stack.Screen name="onboarding" options={{ headerShown: false }} />
            <Stack.Screen name="define-your-goals" options={{ headerShown: false }} />
            <Stack.Screen name="property-detail/[title]" options={{ headerShown: false }} />
          </Stack>
          <PortalHost />
</BottomSheetModalProvider>
```
Now every thing works fine.

**Reply by DannyFung97** [reply]
Following up on this, @saimarshadsaim31, I'm going through the same thing. What did you do to solve this issue?

**Reply by saimarshadsaim31** [reply]
Well it was a long time ago, I did manage to make the select work. You have to have a <PortalHost/> component in the layout but make sure to place it outside of the BottomSheetModalProvider. That worked for me. If that does not work follow this [pattern](https://rnr-docs.vercel.app/getting-started/common-patterns/#a-portal-component-inside-a-presentation-modal-screen). If i recall correctly it did work but had to drop it because of some other gorhom input field component related bugs.
